src/Lykke.Job.BlockchainTransactionsHistoryDetector/Workflow/Processes/NewTransactionDetectionProcess.cs

```diff
@@ -0,0 +1,247 @@
+        private CancellationTokenSource _cancellationTokenSource;
+        private IEventPublisher _eventPublisher;
+        private Task _task;
+
+
+        public NewTransactionDetectionProcess(
+            BlockchainsIntegrationSettings blockchainsIntegrationSettings,
+            IObservableWalletsRepository observableWalletsRepository,
+            ILog log)
+        {
+            // ReSharper disable HeapView.ObjectAllocation.Evident
+            
+            _assetsCache = new OnDemandDataCache<BlockchainAsset>();
+            _blockchainsIntegrationSettings = blockchainsIntegrationSettings;
+            _observableWalletsRepository = observableWalletsRepository;
+            _log = log;
+            
+            // ReSharper enable HeapView.ObjectAllocation.Evident
+        }
+
+
+        public void Dispose()
+        {
+            _cancellationTokenSource?.Cancel();
+
+            try
+            { 
+                _task?.Wait();
+            }
+            catch (AggregateException)
+            {
+
+            }
+        }
+
+        public void Start(ICommandSender commandSender, IEventPublisher eventPublisher)
+        {
+            _cancellationTokenSource = new CancellationTokenSource();
+            _eventPublisher = eventPublisher;
+
+            _task = Task.Run(async () =>
+            {
+                var cancellationToken = _cancellationTokenSource.Token;
+
+                await GetClientsForSupportedBlockchainsAsync();
+                
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    await DetectNewTransactionsAsync(cancellationToken);
+                }
+
+            });
+        }
+
+        private async Task DetectNewTransactionsAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                string continuationToken = null;
+
+                do
+                {
+
+                    IEnumerable<ObservableWalletDto> wallets;
+
+                    (wallets, continuationToken) = await _observableWalletsRepository.GetAllAsync(_blockchainApiClients.Keys, 100, continuationToken);
+
+                    await Task.WhenAll(wallets.Select
+                    (
+                        x => DetectNewTransactionsForWalletAsync(x, cancellationToken)
+                    ));
+
+                } while (continuationToken != null && !cancellationToken.IsCancellationRequested);
+                
+            }
+            catch (Exception e)
+            {
+                await _log.WriteWarningAsync
+                (
+                    component: nameof(NewTransactionDetectionProcess),
+                    process: nameof(DetectNewTransactionsAsync),
+                    context: null,
+                    info: null,
+                    ex: e
+                );
+            }
+
+            await Task.Delay(TimeSpan.FromMinutes(1), cancellationToken);
```

Comment:
Delay between batches could be mutch more smaller. Better to add small (and possibly with little random deviation) delay between each request to the blockchain API and the same delay between outer loop iterations. This will give us more uniform workload distribution without prononced peaks and slumps.